src/borrow_tracker/stacked/mod.rs

```diff
@@ -1181,7 +1014,7 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 // uncovers a non-supported `extern static`.
                 let alloc_extra = this.get_alloc_extra(alloc_id)?;
                 trace!("Stacked Borrows tag {tag:?} exposed in {alloc_id:?}");
-                alloc_extra.stacked_borrows.as_ref().unwrap().borrow_mut().exposed_tags.insert(tag);
+                alloc_extra.borrow_tracker.as_ref().unwrap().expose_tag(tag);
```

Comment:
Why does Stacked Borrows call back into the generic borrow tracking machinery here? This should call `assume_sb`.